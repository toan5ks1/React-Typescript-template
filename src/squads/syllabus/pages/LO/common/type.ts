import { KeyLOTypes } from "src/squads/syllabus/common/constants/const";

import { ActionTypes } from "src/squads/syllabus/hooks/useTopicControl";

export type LOsWithAssignmentType = {
    [x in keyof typeof ActionTypes]: keyof typeof KeyLOTypes | "ASSIGNMENT" | "TASK_ASSIGNMENT";
};

export const losWithAssignmentTypes: LOsWithAssignmentType = {
    ASSIGNMENT: "ASSIGNMENT",
    LO: "LEARNING_OBJECTIVE_TYPE_LEARNING",
    FLASH_CARD: "LEARNING_OBJECTIVE_TYPE_FLASH_CARD",
    OFFLINE_STUDY: "LEARNING_OBJECTIVE_TYPE_OFFLINE_LEARNING",
    EXAM_LO: "LEARNING_OBJECTIVE_TYPE_EXAM_LO",
    TASK_ASSIGNMENT: "TASK_ASSIGNMENT",
};

export interface UpdateLOFormData {
    name: string;
}

export interface CreateLOFormData extends UpdateLOFormData {
    type: LOsWithAssignmentType[keyof LOsWithAssignmentType];
}
