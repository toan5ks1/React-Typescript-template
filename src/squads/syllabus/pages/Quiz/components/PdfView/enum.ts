export enum SelectFieldState {
    RECT_TYPE = "RECT_TYPE",
    FIELD_TYPE = "FIELD_TYPE",
}
