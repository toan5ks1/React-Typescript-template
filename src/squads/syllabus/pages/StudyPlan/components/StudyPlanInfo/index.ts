export { default } from "./StudyPlanInfo";
export * from "./StudyPlanInfo";
