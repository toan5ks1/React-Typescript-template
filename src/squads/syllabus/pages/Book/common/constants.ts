import { convertEnumKeys } from "src/squads/syllabus/common/utils/enum";

import { BookType } from "manabuf/common/v1/contents_pb";

export const BookTypeKeys = convertEnumKeys(BookType);
