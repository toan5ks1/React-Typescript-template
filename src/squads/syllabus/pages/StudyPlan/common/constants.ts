import { convertEnumKeys } from "src/squads/syllabus/common/utils/enum";

import { StudyPlanItemStatus } from "manabuf/eureka/v1/assignments_pb";
import { StudyPlanStatus } from "manabuf/eureka/v1/enums_pb";

import { FormFilterStudyPlanValues, StudyPlanFormData } from "./types";

export const datePattern = /^(\d{4}\/\d{2}\/\d{2})$/;
export const dateTimePattern = /^(\d{4}\/\d{2}\/\d{2},\s\d{2}:\d{2})$/;

export const StudyPlanStatusKey = convertEnumKeys(StudyPlanStatus);
export const StudyPlanItemStatusKey = convertEnumKeys(StudyPlanItemStatus);

export const studyPlanFilterDefaultValues: FormFilterStudyPlanValues = {
    grades: [],
    books: [],
    archived: false,
};

export const defaultStudyPlanFormValues: StudyPlanFormData = {
    name: "",
    book: null,
    grades: [],
    trackSchoolProgress: false,
};
