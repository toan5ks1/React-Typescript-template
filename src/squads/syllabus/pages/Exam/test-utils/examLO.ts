import { KeyLOTypes } from "src/squads/syllabus/common/constants/const";

import { CreateLOFormData } from "src/squads/syllabus/pages/LO/common/type";

const mockCreateLOFormDataEmpty: CreateLOFormData = {
    name: "",
    type: KeyLOTypes.LEARNING_OBJECTIVE_TYPE_EXAM_LO,
    // TODO: "instruction" field will be added later
};

export const createMockExamLOFormData = (overrides = mockCreateLOFormDataEmpty) => {
    return { ...mockCreateLOFormDataEmpty, ...overrides };
};
