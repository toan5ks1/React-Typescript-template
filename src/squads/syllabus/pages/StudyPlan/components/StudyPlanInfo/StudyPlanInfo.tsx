import { Entities } from "src/common/constants/enum";
import { arrayHasItem } from "src/common/utils/other";
import { MicroFrontendTypes } from "src/routing/type";

import { Grid } from "@mui/material";
import StyledLink from "src/components/StyledLink";
import TypographyBase from "src/components/Typographys/TypographyBase";
import TypographyMaxLines from "src/components/Typographys/TypographyMaxLines";

import useConvertGrade from "src/squads/syllabus/hooks/useConvertGrade";
import useResourceTranslate from "src/squads/syllabus/hooks/useResourceTranslate";

export interface StudyPlanInfoProps {
    bookId: string;
    bookName: string;
    grades: Array<number>;
    trackSchoolProgress: boolean;
}

// TODO: This page grid is adjusted with a different grid default
// to match with v4 material-ui grid. We can propose to change to v5 standard
const StudyPlanInfo = (props: StudyPlanInfoProps) => {
    const { bookId, bookName, grades, trackSchoolProgress } = props;
    const tCourses = useResourceTranslate(Entities.COURSES);
    const { convertGradeToString } = useConvertGrade();
    const safeGradeString = arrayHasItem(grades) ? convertGradeToString(grades) : "";

    return (
        <Grid container sx={{ width: "calc(100% + 40px)", margin: "-20px" }}>
            <Grid container item xs={6} p={2.5} sx={{ width: "calc(100% + 16px)", margin: "-8px" }}>
                <Grid item xs={4} p={1}>
                    <TypographyBase variant="body2" color="textSecondary">
                        {tCourses("book")}
                    </TypographyBase>
                </Grid>
                <Grid item xs={8} p={1}>
                    <StyledLink
                        to={`/${MicroFrontendTypes.SYLLABUS}/${Entities.BOOKS}/${bookId}/show`}
                        data-testid="StudyPlanInfo__bookLink"
                    >
                        <TypographyBase variant="body2" color="primary">
                            {bookName}
                        </TypographyBase>
                    </StyledLink>
                </Grid>
                <Grid item xs={4} p={1}>
                    <TypographyBase variant="body2" color="textSecondary">
                        {tCourses("studyPlan.trackSchoolProgress.title")}
                    </TypographyBase>
                </Grid>
                <Grid item xs={8} p={1}>
                    <TypographyMaxLines
                        variant="body2"
                        maxLines={2}
                        data-testid="StudyPlanInfo__trackSchoolProgress"
                    >
                        {tCourses(`studyPlan.trackSchoolProgress.${trackSchoolProgress}`)}
                    </TypographyMaxLines>
                </Grid>
            </Grid>
            <Grid container item xs={6} p={2.5} sx={{ width: "calc(100% + 16px)", margin: "-8px" }}>
                <Grid item xs={4} p={1}>
                    <TypographyBase variant="body2" color="textSecondary">
                        {tCourses("grade")}
                    </TypographyBase>
                </Grid>
                <Grid item xs={8} p={1}>
                    <TypographyMaxLines
                        variant="body2"
                        maxLines={2}
                        data-testid="StudyPlanInfo__grades"
                    >
                        {safeGradeString}
                    </TypographyMaxLines>
                </Grid>
            </Grid>
        </Grid>
    );
};

export default StudyPlanInfo;
