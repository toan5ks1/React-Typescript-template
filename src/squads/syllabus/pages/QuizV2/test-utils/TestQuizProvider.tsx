import { FC } from "react";

import { QuizContext, QuizContextValue } from "../contexts/QuizContext";

import { createTestLO } from "src/squads/syllabus/test-utils/quizV2";

export interface TestQuizProviderProps {
    mode?: QuizContextValue["mode"];
    lo?: QuizContextValue["lo"];
}

const TestQuizProvider: FC<TestQuizProviderProps> = ({
    mode = "create",
    lo = createTestLO(),
    children,
}) => {
    return <QuizContext.Provider value={{ lo: lo, mode }}>{children}</QuizContext.Provider>;
};

export default TestQuizProvider;
