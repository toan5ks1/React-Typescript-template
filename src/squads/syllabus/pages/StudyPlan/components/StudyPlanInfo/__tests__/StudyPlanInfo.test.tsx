import { Entities } from "src/common/constants/enum";
import { MicroFrontendTypes } from "src/routing/type";
import { mockStudyPlanInfo } from "src/squads/syllabus/test-utils/study-plan";

import StudyPlanInfo from "src/squads/syllabus/pages/StudyPlan/components/StudyPlanInfo";

import { fireEvent, render, RenderResult, screen } from "@testing-library/react";
import TestApp from "src/squads/syllabus/test-utils/TestApp";
import TestingRouter from "src/squads/syllabus/test-utils/TestingRouter";

describe("<StudyPlanInfo />", () => {
    let wrapper: RenderResult;

    const gradesString = `Grade ${mockStudyPlanInfo.grades.join(", Grade ")}`;
    const trackSchoolProgressString = mockStudyPlanInfo.trackSchoolProgress ? "Yes" : "No";
    const redirectUrl = `/${MicroFrontendTypes.SYLLABUS}/${Entities.BOOKS}/${mockStudyPlanInfo.bookId}/show`;

    beforeEach(() => {
        wrapper = render(
            <TestApp>
                <TestingRouter redirectUrl={redirectUrl}>
                    <StudyPlanInfo {...mockStudyPlanInfo} />
                </TestingRouter>
            </TestApp>
        );
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });

    it("should render correctly info of <StudyPlanInfo />", () => {
        expect(screen.getByText(mockStudyPlanInfo.bookName)).toBeInTheDocument();
        expect(screen.getByText(gradesString)).toBeInTheDocument();
        expect(screen.getByText(trackSchoolProgressString)).toBeInTheDocument();
    });

    it("should redirect to book detail page when clicking on book name", () => {
        const bookName = screen.getByText(mockStudyPlanInfo.bookName);

        fireEvent.click(bookName);

        expect(screen.getByTestId("TestingRouter__redirectUrl")).toHaveTextContent(redirectUrl);
    });
});
