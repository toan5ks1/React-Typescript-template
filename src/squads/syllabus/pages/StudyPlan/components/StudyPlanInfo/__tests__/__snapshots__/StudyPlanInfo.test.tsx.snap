// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StudyPlanInfo /> should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: calc(100% + 40px);
  margin: -20px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 50%;
  padding: 20px;
  width: calc(100% + 16px);
  margin: -8px;
}

@media (min-width:600px) {
  .emotion-1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 50%;
  }
}

@media (min-width:960px) {
  .emotion-1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 50%;
  }
}

@media (min-width:1440px) {
  .emotion-1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 50%;
  }
}

@media (min-width:1920px) {
  .emotion-1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 50%;
  }
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 33.333333%;
  -ms-flex-preferred-size: 33.333333%;
  flex-basis: 33.333333%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 33.333333%;
  padding: 8px;
}

@media (min-width:600px) {
  .emotion-2 {
    -webkit-flex-basis: 33.333333%;
    -ms-flex-preferred-size: 33.333333%;
    flex-basis: 33.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 33.333333%;
  }
}

@media (min-width:960px) {
  .emotion-2 {
    -webkit-flex-basis: 33.333333%;
    -ms-flex-preferred-size: 33.333333%;
    flex-basis: 33.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 33.333333%;
  }
}

@media (min-width:1440px) {
  .emotion-2 {
    -webkit-flex-basis: 33.333333%;
    -ms-flex-preferred-size: 33.333333%;
    flex-basis: 33.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 33.333333%;
  }
}

@media (min-width:1920px) {
  .emotion-2 {
    -webkit-flex-basis: 33.333333%;
    -ms-flex-preferred-size: 33.333333%;
    flex-basis: 33.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 33.333333%;
  }
}

.emotion-3 {
  margin: 0;
  font-size: 0.875rem;
  font-weight: 400;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.43;
  color: #757575;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 66.666667%;
  -ms-flex-preferred-size: 66.666667%;
  flex-basis: 66.666667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 66.666667%;
  padding: 8px;
}

@media (min-width:600px) {
  .emotion-4 {
    -webkit-flex-basis: 66.666667%;
    -ms-flex-preferred-size: 66.666667%;
    flex-basis: 66.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 66.666667%;
  }
}

@media (min-width:960px) {
  .emotion-4 {
    -webkit-flex-basis: 66.666667%;
    -ms-flex-preferred-size: 66.666667%;
    flex-basis: 66.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 66.666667%;
  }
}

@media (min-width:1440px) {
  .emotion-4 {
    -webkit-flex-basis: 66.666667%;
    -ms-flex-preferred-size: 66.666667%;
    flex-basis: 66.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 66.666667%;
  }
}

@media (min-width:1920px) {
  .emotion-4 {
    -webkit-flex-basis: 66.666667%;
    -ms-flex-preferred-size: 66.666667%;
    flex-basis: 66.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 66.666667%;
  }
}

.emotion-5 {
  cursor: pointer;
  color: #2196F3;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-5:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-6 {
  margin: 0;
  font-size: 0.875rem;
  font-weight: 400;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.43;
  color: #2196F3;
}

.emotion-10 {
  margin: 0;
  font-size: 0.875rem;
  font-weight: 400;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.43;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

<div>
  <div
    class="MuiGrid-root MuiGrid-container emotion-0"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 emotion-1"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-3"
        >
          Book
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 emotion-4"
      >
        <a
          class="emotion-5"
          data-testid="StudyPlanInfo__bookLink"
          href="/syllabus/books/book_id/show"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-6"
          >
            K Book
          </p>
        </a>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-3"
        >
          Track School Progress
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 emotion-4"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-10"
          data-testid="StudyPlanInfo__trackSchoolProgress"
          id="TypographyMaxLines__TypographyBase"
          title=""
        >
          Yes
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 emotion-1"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-3"
        >
          Grade
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 emotion-4"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-10"
          data-testid="StudyPlanInfo__grades"
          id="TypographyMaxLines__TypographyBase"
          title=""
        >
          Grade 5, Grade 6
        </p>
      </div>
    </div>
  </div>
</div>
`;
